creativecloud/features/changeBg/changeBg.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-restricted-syntax */
+let excelLink = '';
+const configObj = {};
+const customElem = document.createElement('ft-changebackgroundmarquee');
+
+async function getExcelData(link) {
+  const resp = await fetch(link);
+  const { data } = await resp.json();
+  return data;
+}
+
+export function getSrcFromExcelData(name, viewportType, excelData, type) {
+  return excelData
+    .filter((data) => data.MenuName === name
+    && data.Viewport === viewportType
+    && data.ResourceName === type)
+    .flatMap((data) => [data.Value1, data.Value2, data.Value3].filter((value) => value.trim() !== ''));
+}
+
+export function createConfigExcel(excelJson, configObjData) {
+  const viewportTypes = ['mobile', 'tablet', 'desktop'];
+  for (const viewportType of viewportTypes) {
+    const existingGroups = configObjData[viewportType].groups;
+    for (const group of existingGroups) {
+      const { name } = group;
+      const groupsrc = getSrcFromExcelData(name, viewportType, excelJson, 'src');
+      const groupswatchSrc = getSrcFromExcelData(name, viewportType, excelJson, 'swatchSrc');
+      group.options = groupsrc.map((src, i) => {
+        const option = { src };
+        if (groupswatchSrc[i]) option.swatchSrc = groupswatchSrc[i];
+        return option;
+      });
+      if (group.options.length === 0) {
+        delete group.options;
+      }
+    }
+  }
+}
+
+export function configExcelData(jsonData) {
+  jsonData.forEach((item) => {
```

Comment:
If we are getting json from excel and keys are there in excel too then can we have a direct mapping of the required config to be generated via franklin json in place of re-mapping?